src/main/events.js

```diff
@@ -286,4 +288,75 @@ export const registerMainProcessCommonEvents = () => {
     const folderDialogPromise = await dialog.showOpenDialog(dialogOptions);
     return folderDialogPromise;
   });
+
+  ipcMain.handle("get-workspace-folder-preview", async (event, payload) => {
```

Comment:
```suggestion
  ipcMain.handle("get-folder-preview", async (event, payload) => {
```
I think we should make this handler generic to return the folder contents for any usecase.